src/GovUk.Education.ExploreEducationStatistics.Data.Model/Migrations/Current_Routine_UpdateStatistics.sql

```diff
@@ -0,0 +1,40 @@
+CREATE OR ALTER PROCEDURE UpdateStatistics
+	@RunId INT
+AS
+BEGIN
+	IF OBJECT_ID('tempdb..#TableList') IS NULL
+    THROW 51000, 'UpdateStatistics expects a temporary table named #TableList to already exist.', 1;  
```

Comment:
This looks like bad practice to me, to depend on a temp table created by another stored procedure.

It's also got another dependency below on `__Log_RebuildIndexesAlterIndexes` so any changes to that in `RebuildIndexes` will break `UpdateStatistics`.

I would have a table value parameter of the tables to update statistics for, iterate through them calling `UPDATE STATISTICS` for each one, and then return.

In the calling SP you could create a variable like `@ModifiedTablesList` and select into it the distinct list of table names which have been modified.

You would then call this SP with something like `EXEC UpdateStatistics @ModifiedTablesList`.